### PR TITLE
Add storage API to data volume

### PR DESCRIPTION
**Pull request by dalia-frank**
A new feature available in 4.10 is storage API
  that will replace the PVC API.
  At the moment, the default API of DataVolume class
  will remain PVC as both available in 4.10.

### CODE REVIEW

File: ocp_resources/datavolume.py [hunk 1]
```diff
@@ -106,40 +107,47 @@ def __init__(
         self.secret = secret
         self.content_type = content_type
         self.size = size
-        self.access_modes = access_modes
+        self.access_modes = (
+            access_modes if use_storage_api or access_modes else self.AccessMode.ROX
+        )
```
Comment (myakove) [inline ocp_resources/datavolume.py:112]:
``` self.access_modes = (
            access_modes if use_storage_api or access_modes else self.AccessMode.ROX
       )
```
Why this is needed? You still use access_mode in both cases.

File: ocp_resources/datavolume.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def __init__(
         privileged_client=None,
         yaml_file=None,
         delete_timeout=TIMEOUT_4MINUTES,
+        use_storage_api=False,
```
Comment (myakove) [inline ocp_resources/datavolume.py:93]:
```suggestion
        storage_api="pvc",
```
I prefer to see here `pvc` as default, and add `TODO` to switch to `storage` when needed or remove it and use only `storage`

File: ocp_resources/datavolume.py [hunk 3]
```diff
@@ -106,40 +107,47 @@ def __init__(
...
         self.source_pvc = source_pvc
         self.source_namespace = source_namespace
         self.multus_annotation = multus_annotation
         self.bind_immediate_annotation = bind_immediate_annotation
         self.preallocation = preallocation
+        self.use_storage_api = use_storage_api
 
     def to_dict(self):
         res = super().to_dict()
+        api_to_use = "storage" if self.use_storage_api else "pvc"
```
Comment (myakove) [inline ocp_resources/datavolume.py:127]: Should be set after line 129, no need to save something that may never be used before needed.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/datavolume.py (hunk 3 above)
Comment (dalia-frank) [reply ocp_resources/datavolume.py:127]: Ack

File: ocp_resources/datavolume.py (hunk 2 above)
Comment (dalia-frank) [reply ocp_resources/datavolume.py:93]: Ack

File: ocp_resources/datavolume.py (hunk 1 above)
Comment (dalia-frank) [reply ocp_resources/datavolume.py:112]: will update so the DV won't have a default value

**Review by myakove — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**